### ISSUE TITLE
[soltest/isoltest] different test results on macOS & Linux

### ISSUE THREAD

**Issue by aarlt** [ethereum/solidity#7468]
`soltest` & `isoltest` currently show different test results on macOS and Linux.

CI should execute tests in the context of macOS.

I did some changes to `.circleci/config.yml` in #7464 to show the different test results in CircleCI, but probably the way I did these changes is not optimal.

**Comment by chriseth** [comment]
Related: #7567

**Comment by aarlt** [comment]
Partially Fixed by #7568. Some tests still fail, because its not easy to change `jsoncpp` (see https://github.com/ethereum/solidity/pull/7568#issuecomment-547367039)
```
solidity/test/libevmasm/Assembler.cpp:148: error: in "Assembler/all_assembly_items": check dev::jsonCompactPrint(_assembly.assemblyJSON()) == "..."
```

@chriseth @Marenz what about fixing this issue by using a case-insensitive compare?

**Comment by aarlt** [comment]
CI now supports the execution of tests within the context of macOS. Closing.
